### PR TITLE
[Android] Upgrade to RN 0.71.0-rc.3

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

* React Native 0.71.0-rc.3
* Android NDK 23

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated

### CODE REVIEW

File: src/android/CMakeLists.txt [hunk 1]
```diff
@@ -28,8 +28,25 @@ target_compile_definitions(realm-js-android PRIVATE
...
+
+    message(STATUS "Uncompressing ${REACT_NATIVE_AAR_DIR}/react-android-0.71.0-rc.3-release.aar")
+    execute_process(
+        COMMAND ${CMAKE_COMMAND} -E tar xfz "react-android-0.71.0-rc.3-release.aar"
+        WORKING_DIRECTORY "${REACT_NATIVE_AAR_DIR}"
+    )
+endif()
+
+# file(GLOB_RECURSE REACT_NATIVE_AAR_FILE ${REACT_NATIVE_ROOT_DIR}/android/*/react-native-*-release.aar)
+# file(ARCHIVE_EXTRACT INPUT ${REACT_NATIVE_AAR_FILE} DESTINATION ${REACT_NATIVE_AAR_DIR})
```
Comment (RedBeard0531) [inline src/android/CMakeLists.txt:49]: Don't forget to delete commented lines

File: src/android/CMakeLists.txt [hunk 2]
```diff
@@ -79,9 +106,10 @@ endif()
 
 # Setup fbjni library
 find_library(FBJNI_LIBRARY fbjni PATHS ${FBJNI_LIB_DIR} NO_CMAKE_FIND_ROOT_PATH)
-target_include_directories(realm-js-android PUBLIC 
+target_include_directories(realm-js-android PUBLIC
     ${FBJNI_INCLUDE_DIR}
     ../../node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/turbomodule/core/jni/
```
Comment (RedBeard0531) [inline src/android/CMakeLists.txt:111]: Based on change on line 14 should this line be removed?

Comment (takameyer) [reply src/android/CMakeLists.txt:112]: I'm not so certain.  Line 14 points to a specific file and this includes a directory.  Although, I could be wrong, as I'm not CMake experienced

Comment (takameyer) [reply src/android/CMakeLists.txt:112]: It does appear to build without.  I'll remove it and see if the tests continue to pass

Comment (takameyer) [reply src/android/CMakeLists.txt:112]: Removing those lines breaks the build.

File: integration-tests/tests/src/tests/sync/sync-as-local.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ describe.skipIf(environment.missingServer, "Synced Realm as local", () => {
   });
 
   before(async function (this: RealmContext) {
-    this.timeout(5000);
+    this.timeout(30 * 1000);
```
Comment (takameyer) [inline integration-tests/tests/src/tests/sync/sync-as-local.ts:36]: We have a global timeout of 60000 set by the GHA configuration.  I'm going to remove these timeouts, as they are not long enough

**Pull request merged**